trace/execution_state.go

```diff
@@ -45,15 +72,18 @@ type WorkflowExecutionState struct {
 
 	// ChildStates contains all the ExecutionStates contained by this WorkflowExecutionState in order of execution.
 	ChildStates []ExecutionState
-	// activityMap contains all the activities executed in the Workflow, indexed by the EVENT_TYPE_ACTIVITY_TASK_SCHEDULED event id.
-	// Used to retrieve the activities from events.
-	activityMap map[int64]*ActivityExecutionState
-	// childWfMap contains all the child workflows executed in the Workflow, indexed by the EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED event id.
-	// Used to retrieve the child workflows from events.
-	childWfMap map[int64]*WorkflowExecutionState
-	// timerMap contains all the timers executed in the Workflow, indexed by the EVENT_TYPE_TIMER_STARTED event id.
-	// Used to retrieve the timers from events.
-	timerMap map[int64]*TimerExecutionState
+	// activityMap contains all the activities executed in the Workflow, indexed by the EVENT_TYPE_ACTIVITY_TASK_SCHEDULED event id
+	// Used to retrieve the activities from events
+	activityMap     map[int64]*ActivityExecutionState
+	activityMapLock sync.RWMutex
```

Comment:
To be safe, I think we should lock the entire struct on any field access.
I haven't looked super closely at the access patterns but since this is all meant to be accessed concurrently, I'd rather be on the safe side.

Reply:
That makes sense, thanks for the suggestion.